huntsman/observatory.py

```diff
@@ -476,6 +481,22 @@ def prepare_cameras(self, sleep=60, max_attempts=5, require_all_cameras=False):
                                   f'waiting another {sleep} seconds before checking again.')
                 time.sleep(sleep)
 
+        # Handle failed cameras
+        if len(cameras_failed) != 0:
+            msg = "The following cameras have failed to become ready: "
+            msg += ", ".join(cameras_failed)
```

Comment:
Might be useful at this point to log why the camera is not ready (temperature unstable, exposing, filterwheel or focuser not ready).